src/app/api/v1/admin/users/[primarySha1]/route.ts

```diff
@@ -109,28 +125,62 @@ export async function PUT(
           ),
         );
 
-        switch (action) {
-          case "subscribe": {
-            // activate and opt out profiles
-            await activateProfile(oneRepProfileId);
-            await optoutProfile(oneRepProfileId);
-            logger.info("force_user_subscribe", {
-              oneRepProfileId,
-              primarySha1,
-            });
-            break;
-          }
-          case "unsubscribe": {
-            await deactivateProfile(oneRepProfileId);
-            logger.info("force_user_unsubscribe", {
-              oneRepProfileId,
-              primarySha1,
-            });
-            break;
-          }
-          default: {
-            logger.error("Unknown action:", action);
-            return NextResponse.json({ success: true }, { status: 500 });
+        for (const action in actions) {
+          switch (action) {
+            case "subscribe": {
+              // activate and opt out profiles
+              await activateProfile(onerepProfileId);
+              await optoutProfile(onerepProfileId);
+              logger.info("force_user_subscribe", {
+                onerepProfileId,
+                primarySha1,
+              });
+              break;
+            }
+            case "unsubscribe": {
+              await deactivateProfile(onerepProfileId);
+              logger.info("force_user_unsubscribe", {
+                onerepProfileId,
+                primarySha1,
+              });
+              break;
+            }
+            case "delete_onerep_profile": {
+              await deleteProfileDetails(onerepProfileId);
+              logger.info("delete_onerep_profile", {
+                onerepProfileId,
+                primarySha1,
+              });
+              break;
+            }
+            case "delete_onerep_scans": {
+              logger.info("delete_onerep_scans", {
+                onerepProfileId,
+                primarySha1,
+              });
+              await deleteScansForProfile(onerepProfileId);
+              break;
+            }
+            case "delete_onrep_scan_results": {
+              await deleteScanResultsForProfile(onerepProfileId);
+              logger.info("delete_onerep_scan_results", {
+                onerepProfileId,
+                primarySha1,
+              });
+              break;
+            }
+            case "delete_subscriber": {
+              await deleteSubscriber(subscriber);
+              logger.info("delete_subscriber", {
+                onerepProfileId,
+                primarySha1,
+              });
+              break;
+            }
+            default: {
+              logger.error("Unknown action:", action);
+              return NextResponse.json({ success: true }, { status: 500 });
```

Comment:
success true?

Reply:
Yes that's not very useful... I saw another place this was true as well.